core/src/avm2/e4x.rs

```diff
@@ -711,8 +742,13 @@ fn to_xml_string_inner<'gc>(xml: E4XOrXml<'gc>, buf: &mut WString) -> Result<(),
             buf.push_str(&escape_element_value(*text));
```

Comment:
`to_xml_string_inner` will never return an Error now, so I think its return type can be removed.
This also means `to_xml_string` (previously returned an `Result<AvmString, Error>`) can now return an `AmString<'gc>`.

Reply:
Good catch!